src/session/session_compact.c

```diff
@@ -260,17 +260,12 @@ __wt_session_compact_check_interrupted(WT_SESSION_IMPL *session)
 
 /*
  * __compact_checkpoint --
- *     This function does wait and force checkpoint.
+ *     This function waits and triggers a checkpoint.
  */
 static int
-__compact_checkpoint(WT_SESSION_IMPL *session)
+__compact_checkpoint(WT_SESSION_IMPL *session, const char *cfg)
```

Comment:
I would isolate the use of config strings inside this function and just have callers use a boolean to indicate whether they need the checkpoint forced. That way the callers don't need to know about config options for checkpoint, and this function doesn't need to worry about whether the caller is providing a reasonable set of options.

Reply:
Good idea! Done in 55153871b1e100ee75b9a980fc46948cc7d44253.